### DISCUSSION TITLE
Helsinki 2021 working - Should the buttons on the display work too?

### DISCUSSION THREAD

**Discussion by heijjan** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#402] [General]
Hi,

And many thanks for this great contribution! I managed to get this working with the V1 board and 10kOhm resistor, although two boards and NodeMCU's were needed in order to get it working, not really sure where it went wrong for the first time. Maybe faulty components, IDK. The display flashed without the resistor at the first try, with the resistor everything rocks.

The remote controls work perfectly and the MQTT integration to Home Assistant is flawless. However I noticed that the pump display's buttons do not do anything anymore, other than beep. I can't see any activities in HA when the physical buttons are pressed and nothing happens. Should they work or not?

**Comment by visualapproach** [comment]
Hi! Thanks!
4-wire: no, when taking control.
6-wire: yes. Check display is unlocked. Press a little longer. 

If it still doesn't work you can make a hardware check. Go to layzspa.local/hwtestinfo.html

**Reply by heijjan** [reply]
Thank you, I have the 6-wire version. Longer press did unlock the screen, but still the buttons only beep, nothing else. I don't have any 5V voltage supply at hand, so testing is a little bit tricky.

**Reply by visualapproach** [reply]
Hi, there is two channels to update ambient. One F and one C. If the device is confused at startup you can try flipping unit back and forth once.
Temperatures is sent on three channels to HA: F, C and whatever you have set as unit. Use F or C as it's units are known. But this should be the default HA config

**Reply by heijjan** [reply]
> Hi, there is two channels to update ambient. One F and one C. If the device is confused at startup you can try flipping unit back and forth once. Temperatures is sent on three channels to HA: F, C and whatever you have set as unit. Use F or C as it's units are known. But this should be the default HA config

I also tried to flip the switch from C to F and back. All the rest readings in HA are correct, I'm updating the ambient value via MQTT command: {"CMD":14,"VALUE":{{ states("sensor.ulkolampotila_layz")|round}},"XTIME":0,"INTERVAL":0}, which sends the data via MQTT as celcius and MQTT returns the value for ambient temp in celcius to HA, but for some reason HA is converting the value and causing too low temperature reading in HA.

**Reply by visualapproach** [reply]
Yes the cmd 14 is for Fahrenheit. Use 15 instead :)

**Reply by heijjan** [reply]
> Yes the cmd 14 is for Fahrenheit. Use 15 instead :)

Haha, alright, thanks. It's interesting how it worked just fine with the older firmware and cmd 14!

**Reply by visualapproach** [reply]
Yes it worked so-so and it could be confused for the F-guys, hence the need for change in behavior.
